src/rgw/rgw_op.cc

```diff
@@ -2481,6 +2481,12 @@ void RGWDeleteBucket::execute()
   }
 
   op_ret = store->delete_bucket(s->bucket, ot);
+
+  if (op_ret == -ECANCELED && !store->is_meta_master()) {
```

Comment:
@cbodley can't this happen with multiple deletes going to the same bucket on the master zone also?